src/main/java/org/gradle/profiler/mutations/FileChangeMutatorConfigurator.java

```diff
@@ -24,30 +24,30 @@ public FileChangeMutatorConfigurator(Class<? extends AbstractFileChangeMutator>
 	public BuildMutator configure(Config scenario, String scenarioName, InvocationSettings settings, String key) {
 		List<BuildMutator> mutatorsForKey = new ArrayList<>();
 		for (File sourceFileToChange : sourceFiles(scenario, scenarioName, settings.getProjectDir(), key)) {
-			mutatorsForKey.add(getBuildMutatorForFile(sourceFileToChange));
+			mutatorsForKey.add(getBuildMutatorForFile(scenario, settings, sourceFileToChange));
 		}
 
 		return new CompositeBuildMutator(mutatorsForKey);
 	}
 
-	private BuildMutator getBuildMutatorForFile(File sourceFileToChange) {
-		if (sourceFileToChange != null) {
-			try {
-				try {
-					return mutatorClass.getConstructor(File.class).newInstance(sourceFileToChange);
-				} catch (InvocationTargetException e) {
-					throw e.getCause();
-				}
-			} catch (Throwable e) {
-				throw new RuntimeException("Could not create instance of mutator " + mutatorClass.getSimpleName(), e);
-			}
-
-		} else {
-			return null;
-		}
+	protected BuildMutator getBuildMutatorForFile(Config scenario, InvocationSettings settings, File sourceFileToChange) {
```

Comment:
This should instead be package private, no? Or just `public`, given that we don't typically use package private in Gradle elsewhere.

Reply:
Changed to private, it stayed protected since I was using it at one point. But reverted it later.